pkg/operator/staticpod/prune/cmd.go

```diff
@@ -0,0 +1,107 @@
+package prune
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"vbom.ml/util/sortorder"
+)
+
+type PruneOptions struct {
+	MaxEligibleRevisionID string
+	ProtectedRevisionIDs  []string
+
+	ResourceDir   string
+	StaticPodName string
+}
+
+func NewPruneOptions() *PruneOptions {
+	return &PruneOptions{}
+}
+
+func NewPrune() *cobra.Command {
+	o := NewPruneOptions()
+
+	cmd := &cobra.Command{
+		Use:   "prune",
+		Short: "Prune static pod installer revisions",
+		Run: func(cmd *cobra.Command, args []string) {
+			glog.V(1).Info(cmd.Flags())
+			glog.V(1).Info(spew.Sdump(o))
+
+			if err := o.Validate(); err != nil {
+				glog.Fatal(err)
+			}
+			if err := o.Run(); err != nil {
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	o.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (o *PruneOptions) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.MaxEligibleRevisionID, "max-eligible-id", o.MaxEligibleRevisionID, "highest revision ID to be eligible for pruning")
+	fs.StringSliceVar(&o.ProtectedRevisionIDs, "protected-ids", o.ProtectedRevisionIDs, "list of revision IDs to reserve from being pruned")
+	fs.StringVar(&o.ResourceDir, "resource-dir", o.ResourceDir, "directory for all files supporting the static pod manifest")
+	fs.StringVar(&o.StaticPodName, "static-pod-name", o.StaticPodName, "name of the static pod")
+}
+
+func (o *PruneOptions) Validate() error {
+	if len(o.ResourceDir) == 0 {
+		return fmt.Errorf("--resource-dir is required")
+	}
+	if len(o.MaxEligibleRevisionID) == 0 {
+		return fmt.Errorf("--max-eligible-id is required")
+	}
+	if len(o.StaticPodName) == 0 {
+		return fmt.Errorf("--static-pod-name is required")
+	}
+
+	return nil
+}
+
+func (o *PruneOptions) Run() error {
+	if err := o.pruneRevisions(); err != nil {
+		return fmt.Errorf("error pruning installerpod resource directory revisions: %v", err)
+	}
+	return nil
+}
+
+func (o *PruneOptions) pruneRevisions() error {
+	protectedIDs := make(map[string]struct{}, len(o.ProtectedRevisionIDs))
```

Comment:
sets.String

Reply:
> protectedDirectories

actually, looking at the logic, you should have `protectedRevisionsId` and then do the separate below in the directory check

Reply:
> sets.String

sets.integer